### ISSUE TITLE
Add binary `if` statement

### ISSUE THREAD

**Issue by sosasees** [godotengine/godot-proposals#8771]
### Describe the project you are working on

a game

### Describe the problem or limitation you are having in your project

sometimes a simple ``if`` statement (like ``if x: y()``) is short enough to fit on 1 line
but this is less readable, so i still put it across 2 lines.

when i try to type it in _binary if_ form (like the _trinary if_ but without the ``else``)
i am disappointed that it just gives an error:
``y() if x``

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

the _binary if_ would be a goodly readable way to put a short ``if`` statement on just 1 line:
instead of
```GDScript
if x:
	y()
```
or the not recommended
```GDScript
if x: y()
```
i could type
```GDScript
y() if x
```

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```GDScript
y() if x
```
would do the same as
```GDScript
if x:
	y()
```

these are examples of how this could be useful:
```GDScript
jump() if Input.is_action_pressed(&'jump')
```
```GDScript
if not score == old_score:
	win() if score >= 172
	lose() if score < 0
```

it's more readable than this:
```GDScript
if Input.is_action_pressed(&'jump'):
	jump()
```
```GDScript
if not score == old_score:
	if score < 0:
		lose()
	elif score >= 172:
		win()
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

no, this is new grammar for GDScript

### Is there a reason why this should be core and not an add-on in the asset library?

if this exists in the asset library, please point me to the add-on, i will use it.
otherwise i believe this is impossible to make as an add-on.

**Comment by Mickeon** [comment]
Do other languages do this? It sounds pretty novel, hence the possible confusion and pushback. But upon second thought, I do kind of like it?

At the end of the day it's an inline `if`, without the following `else`. It feels like a shorthand. One problem is that, inline `if`s expect a value to be returned and assigned to something _(by default the editor warns about this)_. So this usage goes against the convention.

> or the not recommended
> 
> ```gdscript
> if x: y()
> ```

Just because this way is not recommended doesn't mean you can't use it in your code. Because of the reason I explained above, it's unlikely a "Binary If" will be recommended, too.

**Comment by sosasees** [comment]
at first glance
```GDScript
if x: y()
```
looks similar to a variable assignment with static typing:
```GDScript
var x: int = 0
```
. so

```GDScript
y() if x
```
is more readable because it has a distinct if-statement shape

**Comment by dalexeev** [comment]
I agree that in expression context, the incomplete ternary operator is quite confusing. While this makes some sense inside assignments:

```gdscript
var a = 1 if b # else a = null
```

in subexpressions it does not make sense:

```gscript
var a = (1 if b) + 2
```

Note that [null-coalescing operator](https://github.com/godotengine/godot-proposals/issues/1321) does not have this problem, since it has a value for both branches.

**Comment by Mickeon** [comment]
Admittedly, and I just checked, I presumed that this behavior could've replicated by putting `pass` in the `else` condition:

```GDScript
func _ready():
    free() if true else pass
```

But it turns out this is not possible. You need to explicitly pass a blank statement that does nothing.

```GDScript
func _ready():
    free() if true else "Standalone string"
```

One could argue this doesn't make sense. Although, for the `inline if`, both have got to return a Variant, and `pass` does not.

**Comment by KoBeWi** [comment]
> Do other languages do this?

I don't know about other languages, but it exists in Ruby.

**Comment by AThousandShips** [comment]
One issue aside from the syntax confusion from a readability perspective I also suspect this makes the parser for GDScript more complex and sensitive to issues, as it has to handle both options, possibly doing lookahead and risking problems with how it should handle errors and how it should assume what's intended, should something like `foo() if x bar()` be treated as "expected `else` after ..." or something else? Either suggestion would be confusing if the user intends the other syntax, for example what if they intended:
```gdscript
foo() if x
bar()
```
But the compiler assumed they meant:
```gdscript
foo() if x else bar()
```
Then the user would be given the wrong suggestion

**Comment by MewPurPur** [comment]
Adding syntax without purpose to appeal to a few people will make the language as a whole harder to understand. Whether I like the syntax or not, I'll have to learn it to understand others' code.

**Comment by tokengamedev** [comment]
I don't want to bash the intention of proposer, but from me and also coding best practices from professional programmers: _"if you don't handle the else case in a conditional statement, you are bound to create bugs in code/application"_. 

Hence, proceed with caution on this proposal.

**Comment by blockmath** [comment]
> Admittedly, and I just checked, I presumed that this behavior could've replicated by putting `pass` in the `else` condition:
> 
> func _ready():
>     free() if true else pass
> 
> But it turns out this is not possible. You need to explicitly pass a blank statement that does nothing.
> 
> func _ready():
>     free() if true else "Standalone string"
> 
> One could argue this doesn't make sense. Although, for the inline `if`, both have got to return a Variant, and `pass` does not return anything. So, under that logic, it does in fact make sense.

You can just do `free() if true else null`.
